config.go

```diff
@@ -0,0 +1,231 @@
+package main
+
+import (
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ogier/pflag"
+	"github.com/papertrail/remote_syslog2/papertrail"
+	"github.com/papertrail/remote_syslog2/syslog"
+	"github.com/papertrail/remote_syslog2/utils"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	MIN_REFRESH_INTERVAL = RefreshInterval(10 * time.Second)
+	DEFAULT_CONFIG_FILE  = "/etc/log_files.yml"
+)
+
+type Config struct {
+	ConfigFile      string
+	Files           []string
+	DestHost        string
+	DestPort        int
+	Protocol        string
+	Hostname        string
+	RefreshInterval RefreshInterval
+	ExcludeFiles    RegexCollection
+	ExcludePatterns RegexCollection
+	LogLevels       string
+	DebugLogFile    string
+	PidFile         string
+	UseTCP          bool
+	UseTLS          bool
+	Daemonize       bool
+	Severity        syslog.Priority
+	Facility        syslog.Priority
+	Poll            bool
+	RootCAs         *x509.CertPool
+}
+
+type configfile struct {
+	Files       []string
+	Destination struct {
+		Host     string `yaml:"host"`
+		Port     int    `yaml:"port"`
+		Protocol string `yaml:"protocol"`
+	}
+	Hostname        string          `yaml:"hostname"`
+	RefreshInterval RefreshInterval `yaml:"new_file_check_interval"`
+	ExcludeFiles    RegexCollection `yaml:"exclude_files"`
+	ExcludePatterns RegexCollection `yaml:"exclude_patterns"`
+}
+
+func NewConfig() (*Config, error) {
+	self := &Config{
+		ConfigFile:      DEFAULT_CONFIG_FILE,
+		ExcludeFiles:    RegexCollection{},
+		ExcludePatterns: RegexCollection{},
+		DestHost:        "localhost",
```

Comment:
No default value is needed for DestHost, and we'll want to error out if it's not specified in some shape or form.